### PR TITLE
Devdocs: separate UI components from App components

### PR DESCRIPTION
**Pull request by mtias**
- Adds a new section to the sidebar "app components".
- Moves components that rely on application state outside of UI components.
- Fixes missing links to permalinks.

image

### CODE REVIEW

File: client/devdocs/design/app-components.jsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	Sites = require( 'lib/sites-list/docs/example' ),
+	SitesDropdown = require( 'components/sites-dropdown/docs/example' ),
+	Theme = require( 'components/theme/docs/example' ),
+	PostSchedule = require( 'components/post-schedule/docs/example' ),
+	HeaderCake = require( 'components/header-cake' ),
+	Collection,
+	FilterSummary,
+	Hider;
+
+Hider = React.createClass( {
```
Comment (aduth) [inline client/devdocs/design/app-components.jsx]: Future plans to...

- Update to latest syntax?
- Extract components reused between design and app components to separate files?
- Rename `index.jsx` to `ui-components.jsx` ?

Comment (mtias) [inline client/devdocs/design/app-components.jsx]: Yes, all of those. And remove as much of the boilerplate currently needed to add something there.

**Comment by aduth** [comment]
I like this. Adding to above, we should have some more consistency with the route naming:

- `/devdocs/design`
- `/devdocs/app-components`
- `/devdocs/design/typography`

This is a good first step, works well in my testing :+1:

**Comment by mtias** [comment]
I think we should deprecate the `/design` route down the road. I didn't want to break existing links, though.

**Pull request merged**